internal/config/net.go

```diff
@@ -165,12 +165,13 @@ func (t *Net) Opts() []net.DialerOption {
 
 	if t.TLS != nil && t.TLS.Enabled {
 		cfg, err := tls.New(t.TLS.Opts()...)
-		if err == nil {
-			opts = append(opts,
-				net.WithTLS(cfg),
-			)
+		if err != nil {
+			return nil, err
 		}
+		opts = append(opts,
```

Comment:
Like in the internal/config/grpc.go (https://github.com/vdaas/vald/blob/refactor/internal/refactor-config-error-handling/internal/config/grpc.go#L244-L248), does it need to check `cfg` is not nil?

Reply:
it seems to be not needed.
Maybe un-reaach code?
I'll remove it and update test if needed

Reply:
sorry, I misread your comment.
the nil check is not must needed. sorry, my change is not consistent...
if we have to implement a politely code, nil check is better.

Reply:
While searching, I found out every functional option `WithTLS(cfg)` (redis, cassandra, grpc, net) checks whether `cfg` is nil or not.
So, I decided to remove nil check from  https://github.com/vdaas/vald/blob/refactor/internal/refactor-config-error-handling/internal/config/grpc.go#L244-L248.